src/cpp/fastdds/xtypes/utils.cpp

```diff
@@ -0,0 +1,1031 @@
+        const traits<DynamicTypeMember>::ref_type& type_member,
+        nlohmann::json& output,
+        DynamicDataJsonFormat format) noexcept;
+
+static ReturnCode_t json_serialize_member(
+        const traits<DynamicDataImpl>::ref_type& data,
+        MemberId member_id,
+        TypeKind member_kind,
+        const std::string& member_name,
+        nlohmann::json& output,
+        DynamicDataJsonFormat format) noexcept;
+
+static ReturnCode_t json_serialize_member(
+        const traits<DynamicDataImpl>::ref_type& data,
+        MemberId member_id,
+        TypeKind member_kind,
+        nlohmann::json& output,
+        DynamicDataJsonFormat format) noexcept;
+
+static ReturnCode_t json_serialize_basic_member(
+        const traits<DynamicDataImpl>::ref_type& data,
+        MemberId member_id,
+        TypeKind member_kind,
+        const std::string& member_name,
+        nlohmann::json& output,
+        DynamicDataJsonFormat format) noexcept;
+
+static ReturnCode_t json_serialize_collection(
+        const traits<DynamicDataImpl>::ref_type& data,
+        const std::string& member_name,
+        nlohmann::json& output,
+        DynamicDataJsonFormat format) noexcept;
+
+static ReturnCode_t json_serialize_array(
+        const traits<DynamicDataImpl>::ref_type& data,
+        TypeKind member_kind,
+        unsigned int& index,
+        const std::vector<unsigned int>& bounds,
+        nlohmann::json& j_array,
+        DynamicDataJsonFormat format) noexcept;
+
+template <typename T>
+static void json_insert(
+        const std::string& key,
+        const T& value,
+        nlohmann::json& j);
+
+//// Implementation
+
+ReturnCode_t json_serialize(
+        const DynamicData::_ref_type& data,
+        std::ostream& output,
+        DynamicDataJsonFormat format) noexcept
+{
+    ReturnCode_t ret;
+    nlohmann::json j;
+    if (RETCODE_OK == (ret = json_serialize(traits<DynamicData>::narrow<DynamicDataImpl>(data), j, format)))
+    {
+        output << j;
+    }
+    else
+    {
+        EPROSIMA_LOG_WARNING(XTYPES_UTILS, "Error encountered while performing DynamicData to JSON serialization.");
+    }
+    return ret;
+}
+
+ReturnCode_t json_serialize(
+        const DynamicData::_ref_type& data,
+        std::string& output,
+        DynamicDataJsonFormat format) noexcept
+{
+    ReturnCode_t ret;
+    std::stringstream ss;
+
+    if (RETCODE_OK == (ret = json_serialize(data, ss, format)))
+    {
+        output = ss.str();
+    }
+    else
+    {
+        EPROSIMA_LOG_WARNING(XTYPES_UTILS, "Error encountered while performing DynamicData to JSON serialization.");
+    }
+    return ret;
+}
+
+ReturnCode_t json_serialize(
+        const traits<DynamicDataImpl>::ref_type& data,
+        nlohmann::json& output,
+        DynamicDataJsonFormat format) noexcept
+{
+    if (nullptr != data)
+    {
+        switch (data->type()->get_kind())
+        {
+            case TK_STRUCTURE:
+            {
+                ReturnCode_t ret = RETCODE_OK;
```

Comment:
Maybe I'm missing something, why do you define ReturnCode_t ret = RETCODE_OK in here? I would do:
```
DynamicTypeMembersById members;
ReturnCode_t ret = data->type()->get_all_members(members);
if (ret != RETCODE_OK)
{
    EPROSIMA_LOG_WARNING(XTYPES_UTILS, "Error encountered while serializing structure to JSON: get_all_members failed.");
    return ret;
}
```